### PR TITLE
Final updates in documentation and the debian package for v2.20.1

### LINKED ISSUE #1486: Link to installation instructions for macOS missing
**Describe the bug**

In the installation instructions at

https://nest-simulator.readthedocs.io/en/stable/installation/index.html#advanced-install

the reader is referred to the "mac_manual for Macs", see screenshot below, but no link is provided.

**Expected behavior**

A link to the installation instructions for macOS at

https://nest-simulator.readthedocs.io/en/stable/installation/install_options.html#compiling-for-apple-osx-macos

**Screenshots**

image

### PR DESCRIPTION
**Pull request by steffengraber**
This PR changes a few things in the documentation of v2.20.1. Also, the files inside `debian/ ` will be adapted for the new `.deb` package.

fixes #1486

### CODE REVIEW

File: doc/download.rst [hunk 1]
```diff
@@ -40,15 +40,13 @@ Live media is available in the OVA format, and is suitable, for example, for imp
 If you run **Windows**, this is the option for you OR if you just want to run NEST without installing it on your computer.
 After downloading the virtual machine, check out the :doc:`install instructions for Live Media <installation/livemedia>`.
 
-* `NEST Live Media 2.18.0 <https://nest-simulator.org/downloads/gplreleases/lubuntu-18.04_nest-2.18.0.ova>`_
-
-  `Checksum 2.18.0 <https://nest-simulator.org/downloads/gplreleases/lubuntu-18.04_nest-2.18.0.ova.sha512sum>`_
+* `NEST Live Media 2.21.0 <https://nest-simulator.org/downloads/gplreleases/lubuntu-18.04_nest-2.21.0.ova>`_
 
+  `Checksum 2.21.0 <https://nest-simulator.org/downloads/gplreleases/lubuntu-18.04_nest-2.21.0.ova.sha512sum>`_
```
Comment (steffengraber) [inline doc/download.rst]: A new VM will be provided as soon as v2.20.1 is available.

File: doc/download.rst [hunk 2]
```diff
@@ -40,15 +40,13 @@ Live media is available in the OVA format, and is suitable, for example, for imp
 If you run **Windows**, this is the option for you OR if you just want to run NEST without installing it on your computer.
 After downloading the virtual machine, check out the :doc:`install instructions for Live Media <installation/livemedia>`.
 
-* `NEST Live Media 2.18.0 <https://nest-simulator.org/downloads/gplreleases/lubuntu-18.04_nest-2.18.0.ova>`_
-
-  `Checksum 2.18.0 <https://nest-simulator.org/downloads/gplreleases/lubuntu-18.04_nest-2.18.0.ova.sha512sum>`_
+* `NEST Live Media 2.20.1 <https://nest-simulator.org/downloads/gplreleases/lubuntu-18.04_nest-2.20.1.ova>`_
 
+  `Checksum 2.20.1 <https://nest-simulator.org/downloads/gplreleases/lubuntu-18.04_nest-2.20.1.ova.sha512sum>`_
```
Comment (steffengraber) [inline doc/download.rst]: A new VM will be provided as soon as v2.20.1 is available.

**Comment by sarakonradi** [comment]
This PR will also fix #1486.

**Comment by sarakonradi** [comment]
@terhorstd Should we add this to the NEST 2.20.1 Milestone?

**Comment by steffengraber** [comment]
@terhorstd sorry, I pressed the wrong button "requested a review".

File: debian/copyright [hunk 3]
```diff
@@ -1,19 +1,25 @@
...
+
+Files: *
+Copyright: 2004 The NEST Initiative
+License: GPL-2+
 
 Files: debian/*
-Copyright: 2019 The NEST Initiative <users@nest-simulator.org>
-License: GPL-2
+Copyright: 2004 The NEST Initiative
+License: GPL-2+
```
[the remark this answers is not in the document]
Comment (steffengraber) [reply debian/copyright]: You are right, if the two entries are equal, they can be combined. Done

File: debian/control [hunk 4]
```diff
@@ -12,7 +12,7 @@ Standards-Version: 4.1.1
 
 Package: nest
 Architecture: any
-Depends: ${shlibs:Depends}, ${misc:Depends}, python3, libltdl-dev,
+Depends: ${shlibs:Depends}, ${misc:Depends}, python3, libboost-all-dev, libltdl-dev,
```
[the remark this answers is not in the document]
Comment (jougs) [reply debian/control:15]: I agree with @terhorstd  and would also rather go for a minimal set of dependencies.

File: debian/copyright [hunk 5]
```diff
@@ -1,19 +1,21 @@
...
+Source: https://github.com/nest/nest-simulator/
 
-Files: debian/*
-Copyright: 2019 The NEST Initiative <users@nest-simulator.org>
-License: GPL-2
+Files: *
+Copyright: 2004 The NEST Initiative
+License: GPL-2+
+
+License: GPL-2+
```
Comment (jougs) [inline debian/copyright:9]: This line looks redundant to me and I don't understand why the block below is indented. But that might all be just because I don't know anything about the syntax of this file.

File: doc/download.rst [hunk 6]
```diff
@@ -40,15 +40,13 @@ Live media is available in the OVA format, and is suitable, for example, for imp
 If you run **Windows**, this is the option for you OR if you just want to run NEST without installing it on your computer.
 After downloading the virtual machine, check out the :doc:`install instructions for Live Media <installation/livemedia>`.
 
-* `NEST Live Media 2.18.0 <https://nest-simulator.org/downloads/gplreleases/lubuntu-18.04_nest-2.18.0.ova>`_
-
-  `Checksum 2.18.0 <https://nest-simulator.org/downloads/gplreleases/lubuntu-18.04_nest-2.18.0.ova.sha512sum>`_
+* `NEST Live Media 2.20.1 <https://nest-simulator.org/downloads/gplreleases/lubuntu-18.04_nest-2.20.1.ova>`_
```
Comment (jougs) [inline doc/download.rst:43]: Maybe you can discuss the term "Live Media" in the docteam meeting? I really hate it and I think that since CDs are not a thing anymore, we are solely referring to "virtual machine images" if we use that term and should rather just use that term instead. From my perspective, this could, however, also be handled in another PR. Just sayin'.

File: doc/installation/index.rst [hunk 7]
```diff
@@ -241,7 +241,7 @@ these instructions.**
 
    .. tab:: macOS
 
-       For further options on installing NEST on macOS, see :ref:`mac_manual` for Macs.
+       For further options on installing NEST on macOS, see :doc:`mac_install` for Macs.
```
Comment (jougs) [inline doc/installation/index.rst:244]:
```suggestion
       For further options on installing NEST on macOS, see :doc:`mac_install`.
```

File: debian/control (hunk 4 above)
[the remark this answers is not in the document]
Comment (steffengraber) [reply debian/control:15]: @jougs  You mean build NEST with `-Dwith-mpi=OFF `and `-Dwith-boost=OFF`? This would minimize the dependencies.
@terhorstd, what do you think about this?

File: debian/copyright (hunk 5 above)
Comment (steffengraber) [reply debian/copyright:9]: It is a syntax thing.

File: doc/download.rst (hunk 6 above)
Comment (steffengraber) [reply doc/download.rst:43]: > Maybe you can discuss the term "Live Media" in the docteam meeting? I really hate it and I think that since CDs are not a thing anymore, we are solely referring to "virtual machine images" if we use that term and should rather just use that term instead. From my perspective, this could, however, also be handled in another PR. Just sayin'.

@sarakonradi What do you think? I don't really know where the term "live media" comes from. It has been around for quite some time.

File: debian/control (hunk 4 above)
[the remark this answers is not in the document]
Comment (terhorstd) [reply debian/control:15]: building 2.20 with-boost=OFF gives a significant performance hit, so I would suggest against building without it. It really strongly depends on what the deb is going to be used for. If a user builds it to install on his own machine, I'd say with-mpi=OFF is fine. If the deb is built by an admin to roll it out on the cluster we should maybe have a more differentiated explanation. In either case, we should use the minimal with-boost=ON dependencies like libboost-test-dev, libboost-container-dev, libboost-tools-dev, etc. (not sure if these ones suffice, I may be missing a packet)

[the remark this answers is not in the document]
Comment (steffengraber) [reply debian/control:15]: > we should use the minimal with-boost=ON dependencies like libboost-test-dev, libboost-container-dev, libboost-tools-dev, etc. (not sure if these ones suffice, I may be missing a packet)

@jougs Are these dependencies for boost okay? Or is something missing?

File: doc/download.rst (hunk 6 above)
Comment (jougs) [reply doc/download.rst:43]: The old term was introduced by me somewhere around 2004, when the new big thing were _Live CDs_ or _bootable USB drives_. The unifying term for the ISO images usable on both types of media was officially _Live Meda_. Being young and naïve, I thoughtlessly took that over, but think it is about time to drop it.

File: debian/copyright (hunk 5 above)
Comment (jougs) [reply debian/copyright:9]: Alright, then forget about my comment.

File: debian/control (hunk 4 above)
[the remark this answers is not in the document]
Comment (jougs) [reply debian/control:15]: I agree with what @terhorstd said in his last comment.

@hakonsbm: can you comment on the list of boost-related libraries we need? I was not involved in that and thus can't say a whole lot about it.

[the remark this answers is not in the document]
Comment (hakonsbm) [reply debian/control:15]: @jougs We need Boost for sorting and to compile the C++ tests. There is a separate `libboost-test` package, but I don't find a separate package for the sorting. So for a package based installation I think `libboost-all-dev` is the only option.

However, when #1502 is merged it will also be possible to download the Boost headers only and compile with those.

[the remark this answers is not in the document]
Comment (steffengraber) [reply debian/control:15]: @hakonsbm Thanks!
Now we have `-Dwith-mpi=OFF` and `libboost-all-dev` stays as it is.

[the remark this answers is not in the document]
Comment (steffengraber) [reply debian/control:15]: @terhorstd what's your opinion?

[the remark this answers is not in the document]
Comment (terhorstd) [reply debian/control:15]: On Travis we use separate boostlibs and those are the tested and obviously working ones. I think we should take those and not force libboost-all-dev. The sorting is probably in the minimal `libboost-dev` and not in a separate `libboost-*-dev`

**Comment by terhorstd** [comment]
@steffengraber, is there an update on the separate `boost-*-dev` libs used by travis? If not so, we maybe should open another issue to minimize requirements and get this one through for now.

File: debian/control (hunk 4 above)
[the remark this answers is not in the document]
Comment (terhorstd) [reply debian/control:15]: After discussion, we decided to split this to a separate issue
